### PR TITLE
Show major browser version on first load of Test Run

### PR DESCRIPTION
**Pull request by evmiguel**
This PR addresses #810, which aims to only show the major browser version when confirming the user's browser version.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206550059310407

### CODE REVIEW

File: client/components/common/AtAndBrowserDetailsModal/index.jsx [hunk 1]
```diff
@@ -90,15 +95,13 @@ const AtAndBrowserDetailsModal = ({
         }
 
         const foundBrowserVersion =
-            uaBrowser === browserName &&
-            uaMajor !== '0' &&
-            `${uaMajor}.${uaMinor}.${uaPatch}`;
+            uaBrowser === browserName && uaMajor !== '0' && `${uaMajor}`;
```
Comment (howard-e) [inline client/components/common/AtAndBrowserDetailsModal/index.jsx:98]:
```suggestion
            uaBrowser === browserName && uaMajor !== '0' && uaMajor;
```

File: client/components/common/AtAndBrowserDetailsModal/index.jsx [hunk 2]
```diff
@@ -90,15 +95,13 @@ const AtAndBrowserDetailsModal = ({
...
 
         if (
             // don't force browserVersion update with admin (unless first run)
             (!isAdmin || (isAdmin && firstLoad)) &&
             // check that saved browserVersion is the same as detected
-            !browserVersion.includes(`${uaMajor}.${uaMinor}.${uaPatch}`) &&
+            !browserVersion.includes(`${uaMajor}`) &&
```
Comment (howard-e) [inline client/components/common/AtAndBrowserDetailsModal/index.jsx:104]: Leaving as the string template does make it more intentional but the hook does return `ua...` values as string if you'd like to simplify this.

The function would also implicitly convert a numeric value to string as well. But this is a nit!

```suggestion
            !browserVersion.includes(uaMajor) &&
```

**Review by howard-e — COMMENTED** [review]
Thanks @evmiguel! Everything looks good and works as expected.

I think you could also include a change to the displayed browser version at the top of the Test Run page as well. See screenshot.

screenshot showing full version number information for browser at top of test run page, as Chrome 120.0.0

File: client/components/common/AtAndBrowserDetailsModal/index.jsx [hunk 3]
```diff
@@ -194,7 +194,26 @@ const AtAndBrowserDetailsModal = ({
                 );
         }
 
-        handleAction(updatedAtVersion, updatedBrowserVersion, updateMessage);
+        // Save full version even though we don't display it.
+        // Discard trailing zeros in found minor and patches.
+        // This also accounts for trailing numbers after the patch.
+        let fullVersion = '';
+        let [majorVersion, minorVersion, patch, ...rest] =
```
Comment (howard-e) [inline client/components/common/AtAndBrowserDetailsModal/index.jsx:201]:
```suggestion
        let [majorVersion, minorVersion = '0', patch = '0', ...rest] =
```

Also, could you use `patchVersion` for consistency?

File: client/components/common/AtAndBrowserDetailsModal/index.jsx [hunk 4]
```diff
@@ -194,7 +194,26 @@ const AtAndBrowserDetailsModal = ({
...
+            updatedBrowserVersion.split('.');
+
+        if (minorVersion === '0' && (patch === '0' || patch === undefined)) {
```
Comment (howard-e) [inline client/components/common/AtAndBrowserDetailsModal/index.jsx:204]:
```suggestion
        if (minorVersion === '0' && patch === '0') {
```

The `minorVersion` being `undefined` wasn't being handled which was causing cases of `{MAJOR}.undefined` being saved to the database.

File: client/components/common/AtAndBrowserDetailsModal/index.jsx [hunk 5]
```diff
@@ -194,7 +194,26 @@ const AtAndBrowserDetailsModal = ({
...
+            fullVersion = `${majorVersion}`;
+        } else if (patch === '0' || patch === undefined) {
```
Comment (howard-e) [inline client/components/common/AtAndBrowserDetailsModal/index.jsx:206]:
```suggestion
        } else if (patch === '0') {
```

**Review by howard-e — COMMENTED** [review]
@evmiguel looks good, thanks for taking a look at this! I've left some inline comments and suggestions before approving.

**Review by howard-e — APPROVED** [review]
@evmiguel looks good! Thanks for addressing my feedback

**Comment by alflennik** [comment]
Hi @evmiguel thanks for sorting out the question I raised about how to tackle this issue! 

So I created a new entry in the test queue using chrome and it detected and displayed a version of 122, which is perfect. However when I looked at the database it did not store the minor or patch parts of the version number. Is that expected? See below for a screenshot of the database contents.

Table of browser versions in the database which include `99.0.4844.84` and `122` for chrome

If we plan to store the full version number, which I think we do, I think this might be a bug worth fixing. If we're only planning on storing the major number I think we should update our automatically seeded data to only include the major number.

The full version number, by the way, was 122.0.0, I got that number like this:

Code reading console.log(`${uaMajor}.${uaMinor}.${uaPatch}`)

**Comment by alflennik** [comment]
Ah ok, sounds good then!

**Review by alflennik — APPROVED** [review]

**Pull request merged**